server/deadfish.cpp

```diff
@@ -0,0 +1,76 @@
+#include "deadfish.hpp"
+#include "../common/constants.hpp"
+
+// HidingSpot
+
+HidingSpot::HidingSpot(const FlatBuffGenerated::HidingSpot* fb_Hs) {
+    this->name = fb_Hs->name()->str();
+    b2BodyDef myBodyDef;
+	myBodyDef.type = b2_staticBody;
+	myBodyDef.position.Set(fb_Hs->pos()->x(), fb_Hs->pos()->y());
+	body = gameState.b2world->CreateBody(&myBodyDef);
+
+	b2FixtureDef fixtureDef;
+    fixtureDef.isSensor = true; // makes hiding spot detect collision but allow movement
+    if (fb_Hs->ellipse()) {
+        b2CircleShape circleShape;
+        circleShape.m_radius = fb_Hs->radius();
+        fixtureDef.shape = &circleShape;
+    } else {
+        b2PolygonShape polyShape;
+        std::vector<b2Vec2> vertices;
+        auto fb_poly = fb_Hs->polyverts();
+        for (auto vert = fb_poly->begin(); vert != fb_poly->end(); ++vert) {
+            vertices.push_back(b2Vec2(vert->x(), vert->y()));
+        }
+        polyShape.Set(vertices.data(), vertices.size());
+        fixtureDef.shape = &polyShape;
+    }
+	fixtureDef.filter.categoryBits = 0x0010;    // "i be bush"
+    fixtureDef.filter.maskBits = 0x0002;        // "i collide with player"
+	body->CreateFixture(&fixtureDef);
+	body->SetUserData(this);
+}
+
+void HidingSpot::handleCollision(Collideable& other) {
+    try {
+        auto &player = dynamic_cast<Player &>(other);
+        playersInside.insert(&player);
+    }
+    catch (...) {
+        std::cout << "non-player collision with bush detected!" << std::endl;
```

Comment:
If I understand correctly, this is an unnecessary debug, if so, please remove it. The server output is too cluttered as it is

Reply:
This only happens if something really goes wrong, it doesn't show up if the server is working properly. Figured it could stay in case something breaks. Can be removed, but left intentionally

Reply:
Won't this show up every time a mob enters a hiding spot?

Reply:
Nope, mob-hspot collisions are filtered by box2d. It shouldn't show up at all; if it does, that means that somebody configured something wrong regarding box2d, collision masks or something like that.